src/distributions/uniform.rs

```diff
@@ -279,7 +298,7 @@ pub trait UniformSampler: Sized {
         B1: SampleBorrow<Self::X> + Sized,
         B2: SampleBorrow<Self::X> + Sized,
     {
-        let uniform: Self = UniformSampler::new(low, high);
+        let uniform: Self = UniformSampler::new(low, high).unwrap();
```

Comment:
Why does this unwrap instead of forwarding the error?

Reply:
Because `sample_single` (and by extension `gen_range`) don't return errors.

Reply:
`gen_range` documents that it panics, and anyway, all `Rng` methods are just sugar, so fine.

`Uniform::sample` has no need for error handling, good.

But `sample_single{,_inclusive}` are the base implementations and do need error handling, so it seems like they should return a `Result` too.

Reply:
I made it so.